treeherder/auth/backends.py

```diff
@@ -85,9 +100,12 @@ def authenticate(self, authorization=None, host=None, port=None):
         # TODO: remove this size limit when we upgrade to django 1.10
         # in Bug 1311967
         email = self._get_email(result)
-        sha = sha1()
-        sha.update(email)
-        username = sha.hexdigest()[-30:]
+        if len(result["clientId"]) <= 30:
+            username = result["clientId"]
```

Comment:
We should still always use the email for the username rather than `result["clientId"]`, the difference is that we only have to hash that email if it's >30 characters :-)